pkg/virt-launcher/notify-client/client.go

```diff
@@ -196,12 +196,13 @@ func newWatchEventError(err error) watch.Event {
 }
 
 func eventCallback(c cli.Connection, domain *api.Domain, libvirtEvent libvirtEvent, client *Notifier, events chan watch.Event,
-	interfaceStatus []api.InterfaceStatus, osInfo *api.GuestOSInfo) {
+	interfaceStatus []api.InterfaceStatus, osInfo *api.GuestOSInfo, domainName string) {
 	d, err := c.LookupDomainByName(util.DomainFromNamespaceName(domain.ObjectMeta.Namespace, domain.ObjectMeta.Name))
 	if err != nil {
 		if !domainerrors.IsNotFound(err) {
 			log.Log.Reason(err).Error("Could not fetch the Domain.")
-			client.SendDomainEvent(newWatchEventError(err))
+			errMessage := fmt.Errorf(fmt.Sprintf("Could not fetch the Domain %s, reason %s", domainName, err.Error()))
```

Comment:
`Errorf` is supposed to be similar to `Printf`, so I do not think you need `Sprintf`.
Also, I saw errors being formatted using %v and without `Error()` (e.g. `fmt.Errorf("%v", err)`)

The name is also not representing what this is. It is not a message, it is an error. You could call it something like `canNotFetchDomainError`.

Same for the others below.